pkg/virt-api/api_test.go

```diff
@@ -185,128 +324,415 @@ var _ = Describe("Virt-api", func() {
 		}, 5)
 
 		It("should create apiservice endpoint if one doesn't exist", func(done Done) {
-			app.signingCertBytes = []byte("fake")
-			apiService := &apiregistrationv1beta1.APIService{
-				ObjectMeta: metav1.ObjectMeta{
-					Name: subresourceAggregatedApiName,
-					Labels: map[string]string{
-						v1.AppLabel: "virt-api-aggregator",
-					},
-				},
-				Spec: apiregistrationv1beta1.APIServiceSpec{
-					Service: &apiregistrationv1beta1.ServiceReference{
-						Namespace: namespaceKubevirt,
-						Name:      "virt-api",
-					},
-					Group:                v1.SubresourceGroupName,
-					Version:              v1.SubresourceGroupVersion.Version,
-					CABundle:             app.signingCertBytes,
-					GroupPriorityMinimum: 1000,
-					VersionPriority:      15,
-				},
-			}
+			expectedApiService := app.subresourceApiservice()
```

Comment:
I don't think we should be using api.go code to generate the expected ApiService here. Just like we have a pre-configured expected validation and mutation webhook, we should have a pre-configured api service that we're validating against.

The point here is to catch any unexpected alteration in the creation of the apiService object. If api.go changes how the apiService is generated, then the test should fail until it is explicitly updated to reflect that change.

Reply:
So you mean that we have to replicate ApiServce and validation and mutation hook config in the test ? I didn't want to do that to reduce maintenance, was just testing the create/update failure cycle.

Reply:
we need to validate the app.subresourceApiserice() generation code. Even though the previous test was flawed, part of the intent was to ensure that the service generation code produced results we expect.